#### Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [19]:
import pandas as pd

data = [[8,9,10,7,5],[90,92,95,90,85]]
data_transposed = list(zip(*data))
df = pd.DataFrame(data_transposed,columns=['Time_spent_studying','Final_score'])
df.corr(method = 'pearson')

Time_spent_studying  Final_score
Time_spent_studying             1.000000     0.976481
Final_score                     0.976481     1.000000

The results shows that Time spent studying has positive and strong correlation with the final score as 0.976481 is very close to positive 1.

#### Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [24]:
import pandas as pd

data = [[8,9,6,7,5],[9,8,7,8,5]]
data_transposed = list(zip(*data))
df = pd.DataFrame(data_transposed,columns=['Time_spent_sleeping','Satisfaction_level'])
df.corr(method = 'spearman')

Time_spent_sleeping  Satisfaction_level
Time_spent_sleeping             1.000000            0.820783
Satisfaction_level              0.820783            1.000000

The results shows that Time spent sleeping has positive and strong correlation with the final score as 0.820783 is very close to positive 1.

#### Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [43]:
import pandas as pd

data = [[7,3.5,14,21,0],[22.5,24,22,21,28]]
data_transposed = list(zip(*data))
df = pd.DataFrame(data_transposed,columns=['Hour_spent_exercising','BMI'])
df.corr(method = 'spearman')

Hour_spent_exercising  BMI
Hour_spent_exercising                    1.0 -1.0
BMI                                     -1.0  1.0

In [27]:
df.corr(method = 'pearson')

Hour_spent_exercising      BMI
Hour_spent_exercising                1.00000 -0.85286
BMI                                 -0.85286  1.00000

The results shows that Time spent exercising has negative but strong correlation with the BMI as -1 (spearman correlation) is perfect negative correlation and -0.85386 (pearson correlation) is very close to negative 1.

#### Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [42]:
import numpy as np
np.random.seed(42)
hours_tv = np.random.normal(4, 1.5, 50) 
physical_activity = np.random.normal(6, 2, 50)
data = [hours_tv,physical_activity]
data_transposed = list(zip(*data))
df = pd.DataFrame(data_transposed,columns=['Hour_spent_on_TV','Physical_activity'])
df.corr(method = 'pearson')


Hour_spent_on_TV  Physical_activity
Hour_spent_on_TV           1.000000           0.110072
Physical_activity          0.110072           1.000000

The results shows that Time spent watching television per day has near to no correlation with physical activity as 0.110072 is close to 0.

#### Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:
|Age(Yrs)|Soft drink preference|
|--|--|
|25|Coke|
|42|Pepsi|
|37|Mountain Dew|
|19|Coke|
|31|Pepsi|
|28|Coke|

Since the question is incomplete, I could not answer it.

#### Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [41]:
import numpy as np

np.random.seed(42)

sales_calls_per_day = np.random.randint(30, 60, 30)
sales_per_week = np.random.randint(5, 15, 30)

data = [sales_calls_per_day,sales_per_week]
data_transposed = list(zip(*data))
df = pd.DataFrame(data_transposed,columns=['sales_calls_per_day','sales_per_week'])
df.corr(method = 'pearson')


sales_calls_per_day  sales_per_week
sales_calls_per_day             1.000000       -0.023002
sales_per_week                 -0.023002        1.000000

The results shows that Sales Calls per day has near to no correlation with Sales per week as -0.023002 is close to 0.